Sources/SwiftInspectorVisitors/NestableDeclSyntax.swift

```diff
@@ -0,0 +1,34 @@
+// Created by Dan Federman on 2/15/21.
+//
+// Copyright © 2021 Dan Federman
+//
+// Distributed under the MIT License
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+import SwiftSyntax
+
+protocol NestableDeclSyntax: SyntaxProtocol {
+  var identifier: TokenSyntax { get }
+  var modifiers: ModifierListSyntax? { get }
+}
+
+extension ClassDeclSyntax: NestableDeclSyntax {}
+extension StructDeclSyntax: NestableDeclSyntax {}
+extension EnumDeclSyntax: NestableDeclSyntax {}
```

Comment:
These types all have _very_ similar properties available on them. I don't foresee any issues combining them into a single protocol.

Here they are side-by-side for comparison (with comments to help guide differences):
```
public struct ClassDeclSyntax: DeclSyntaxProtocol, SyntaxHashable {
  enum Cursor: Int {
    case attributes
    case modifiers
    case classKeyword // this keyword is unique to this type, but all it says is "class" which we can get from the type information.
    case identifier
    case genericParameterClause
    case inheritanceClause
    case genericWhereClause
    case members
  }
...
}

public struct StructDeclSyntax: DeclSyntaxProtocol, SyntaxHashable {
  enum Cursor: Int {
    case attributes
    case modifiers
    case structKeyword // this keyword is unique to this type, but all it says is "class" which we can get from the type information.
    case identifier
    case genericParameterClause
    case inheritanceClause
    case genericWhereClause
    case members
  }
 ...
}

public struct EnumDeclSyntax: DeclSyntaxProtocol, SyntaxHashable {
  enum Cursor: Int {
    case attributes
    case modifiers
    case enumKeyword // this keyword is unique to this type, but all it says is "class" which we can get from the type information.
    case identifier
    case genericParameters
    case inheritanceClause
    case genericWhereClause
    case members
  }
 ...
}
```